doc/source/specification.rst

```diff
@@ -1,11 +1,95 @@
 
-Soma formats:
--------------
-MorphIO implements Soma specification from `NeuroMorpho`_, and recognizes several kinds of soma format.
+
+Point
+-----
+A point is an array of three numbers **[X, Y, Z]** where X, Y, Z are Cartesian coordinates of position.
+A point always has a diameter associated with it.
+
+
+Segment
+-------
+A segment consists of two consecutive points(`Point`_) belonging to the same section(`Section`_).
+
+.. image:: images/swc/two_pt.svg
+   :scale: 100 %
+   :alt: segment
+
+
+Section
+-------
+
+A section is a series of one or more segments. Each section has a type associated with it. The type shows what part of
+the neuron a section represents. The type can be the axon, the soma, and so on.
+`Section API <cpp/doxygen_Section.html>`_.
+
+.. image:: images/section_segment.svg
+   :scale: 100 %
+   :alt: section
+
+The first and the last point of section must be of the following combinations:
+
+1. starting point, forking point
+2. forking point, end point
+3. forking point, forking point
+4. starting point, end point
+
+Starting point is the first point of a section without parent. Forking point is the last point of a section that has
+multiple children. End point is the last point of a section without children. For more details on section's parent and
+children see `Neurite`_. The above combinations are marked with their corresponding number at the image below.
+
+.. image:: images/sections.svg
+   :scale: 100 %
+   :alt: section variants
+
+Neurite
+-------
+
+A neurite is essentially a tree of sections(`Section`_). The tree structure implies the following:
+
+* Section can have only one parent (another section).
+* Section can have an arbitrary number of children (other sections).
+* No loops are present in the structure.
+
+A section without parent is called a root section. A section with parent must have its first point to be a duplicate
+of the last point of its parent.
+
+Section ordering
+****************
+In MorphIO each section is identified by an ID. By default, the section IDs will correspond to
+the order of section appearance while performing a depth-first traversal on every neurites. The
+neurite order is the order of appearance in the file. Alternatively, the NRN simulator way of
+ordering section can be used by specifying the flag ``morphio::Option::NRN_ID`` when opening
+the file. In the NRN simulator, the soma which is considered as a section (contrary to MorphIO)
+is placed first and then neurites are sorted according to their type.
+
+The final order is the following:
+
+0. Soma
+1. Axon
+2. Basal
+3. Apical
+
+Section with only one child section
+***********************************
+Prior to version 3.0.0, when a section had a single child section (aka unifurcation), the child section would be merged
+with its parent when reading or writing the file. Since version 3.0.0, merging does not happen when reading. Yet
+writing of such sections is not allowed.
```

Comment:
could you point to `remove_unifurcation` method? If you try to save, what happens, this is called? I'm not sure I tried to save with unifurcations, but I find this weird that we cannot save these as they are.

Reply:
done